File: frontend/lib/data/models/UserModel.dart
class UserModel {
  String? _id;
  String? _firstName;
  String? _lastName;
  String? _email;
  String? _profilePicture;

  UserModel({
    String? id,
    String? firstName,
    String? lastName,
    String? email,
    String? profilePicture,
  }) {
    _id = id;
    _firstName = firstName;
    _lastName = lastName;
    _email = email;
    _profilePicture = profilePicture;
  }

  String? get id => _id;
  String? get firstName => _firstName;
  String? get lastName => _lastName;
  String? get email => _email;
  String? get profilePicture => _profilePicture;

  UserModel.fromJson(Map<String, dynamic> json) {
    _id = json['_id'];
    _firstName = json['firstName'];
    _lastName = json['lastName'];
    _email = json['email'];
    _profilePicture = json['profilePicture'];
  }

  Map<String, dynamic> toJson() {
    return {
      'id': _id,
      'firstName': _firstName,
      'lastName': _lastName,
      'email': _email,
      'profilePicture': _profilePicture,
    };
  }
}

File: backend/src/users/user.model.ts
import * as mongoose from 'mongoose';

export const UserSchema = new mongoose.Schema({
  firstName: { type: String, required: true },
  lastName: { type: String, required: true },
  email: { type: String, required: true },
  password: { type: String, required: true },
  profilePicture: { type: String, required: false },
});

export interface User extends mongoose.Document {
  id: string;
  firstName: string;
  lastName: string;
  email: string;
  password: string;
  profilePicture: string;
}
